### ISSUE TITLE
Memory leaks in floating-point compression code

### ISSUE THREAD

**Issue by PhilMiller** [UIUC-PPL/charm#271]
*Original issue: https://charm.cs.illinois.edu/redmine/issues/271*

---

```
[src/arch/util/compress.c:218]: (error) Memory leak: baseData
[src/arch/util/compress.c:274]: (error) Memory leak: baseData
[src/arch/util/compress.c:499]: (error) Memory leak: baseData
```

The leaks all result from the case where `bData` is `NULL` on input, and so memory is allocated, but the returned pointer never escapes the individual function.

**Comment by pplimport** [comment]
*Original author: Yanhua Sun*
*Original date: 2013-08-26 19:41:34*

---
Phil, 

I checked and noticed when bData is NULL, that function should not be called at all.  How did you detect the memory leak?

**Comment by pplimport** [comment]
*Original author: Yanhua Sun*
*Original date: 2013-08-26 21:56:31*

---
added assert(bData != NULL)  to avoid the memory leak. Checked in

**Comment by PhilMiller** [comment]
*Original date: 2013-08-28 18:54:47*

---
Fixed in commit:774d2cd18d416bd1f8d4b5849330bfe00c03559f
